src/shout_subgroup/repository.py

```diff
@@ -110,6 +111,43 @@ async def insert_subgroup(
     return new_subgroup
 
 
+async def delete_subgroup(
+        db: Session,
+        telegram_group_chat_id: int,
+        subgroup_name: str,
+) -> bool:
+    """
+    Hard deletes a subgroup from the database.
+    This delete is a cascading delete, meaning
+    if there are users associated with the subgroup,
+    the associations will be removed prior to the deletion.
+    :param db:
+    :param telegram_group_chat_id:
+    :param subgroup_name:
+    :return:
+    """
+
+    subgroup = await find_subgroup_by_telegram_group_chat_id_and_subgroup_name(
+        db,
+        telegram_group_chat_id,
+        subgroup_name
+    )
+
+    if not subgroup:
+        return False
+
+    # This is a cascading delete.
```

Comment:
Focus: @rick1e and I decided to do a hard delete instead of a soft delete. We'll remove the users in the subgroup, then delete the subgroup